narwhal/network/src/connectivity.rs

```diff
@@ -69,20 +77,38 @@ impl ConnectionMonitor {
         }
 
         // now report the connected peers
-        for peer_id in connected_peers {
-            self.handle_peer_status_change(peer_id, ConnectionStatus::Connected)
+        for peer_id in connected_peers.iter() {
+            self.handle_peer_status_change(*peer_id, ConnectionStatus::Connected)
                 .await;
         }
 
-        while let Ok(event) = subscriber.recv().await {
-            match event {
-                anemo::types::PeerEvent::NewPeer(peer_id) => {
-                    self.handle_peer_status_change(peer_id, ConnectionStatus::Connected)
-                        .await;
+        let mut connection_stat_collection_interval = time::interval(Duration::from_secs(60));
+
+        loop {
+            tokio::select! {
+                _ = connection_stat_collection_interval.tick() => {
```

Comment:
Do we care that this can't exit anymore? Will this create any problems on shutdown/reconfig?

Reply:
This is what happens when I decide I am too cool to run tests locally 🤦🏽 . The nw reconfig tests are failing in CI, will fix

Reply:
Yes we care :)